test/regress/cli/run_regression.py

```diff
@@ -174,11 +174,106 @@ def applies(self, benchmark_info):
+                "unsat" in benchmark_info.expected_output.split()
+                or "entailed" in benchmark_info.expected_output.split()
+            )
+            and "-i" not in benchmark_info.command_line_args
+            and "--incremental" not in benchmark_info.command_line_args
+            and "--no-produce-proofs" not in benchmark_info.command_line_args
+            and "--no-check-proofs" not in benchmark_info.command_line_args
+            and "--check-proofs" not in benchmark_info.command_line_args
+            and "--produce-unsat-cores" not in benchmark_info.command_line_args
+            and "--dump-unsat-cores" not in benchmark_info.command_line_args
+            and ":incremental true" not in benchmark_info.benchmark_content
+            and ":produce-proofs false" not in benchmark_info.benchmark_content
+            and ":check-proofs" not in benchmark_info.benchmark_content
+            and ":produce-unsat-cores true" not in benchmark_info.benchmark_content
+            and ":dump-unsat-cores true" not in benchmark_info.benchmark_content
+            and "(reset)" not in benchmark_info.benchmark_content
+        )
+
+    def run(self, benchmark_info):
+        exit_code = None
+        tmpf_name = None
+        with tempfile.NamedTemporaryFile(delete=False) as tmpf:
+            proof_args = [
+                "--dump-proofs",
+                "--proof-format=lfsc",
+                "--proof-granularity=theory-rewrite",
+            ]
+            cvc5_output, cvc5_error, exit_status = run_process(
+                [benchmark_info.cvc5_binary]
+                + benchmark_info.command_line_args
+                + proof_args
+                + [benchmark_info.benchmark_basename],
+                benchmark_info.benchmark_dir,
+                benchmark_info.timeout,
+            )
+            tmpf_name = tmpf.name
+            tmpf.write(cvc5_output.strip("unsat\n".encode()))
+        if not tmpf_name:
+            exit_code = EXIT_FAILURE
+            print("not ok (cvc5) - Flags: {}".format(benchmark_info.command_line_args))
+        elif exit_status == STATUS_TIMEOUT:
+            exit_code = EXIT_SKIP if g_args.skip_timeout else EXIT_FAILURE
+            print("Timeout (cvc5) - Flags: {}".format(benchmark_info.command_line_args))
+        elif exit_status != 0:
+            exit_code = EXIT_FAILURE
+            print(
+                'not ok (cvc5) - Expected exit status 0 but got "{}" - Flags: {}'.format(
+                    exit_status,
+                    benchmark_info.command_line_args,
+                )
+            )
+            print()
+            print("Output:")
+            print("=" * 80)
+            print_colored(Color.BLUE, cvc5_output.decode())
+            print("=" * 80)
+            print()
+            print()
+            print("Error output:")
+            print("=" * 80)
+            print_colored(Color.YELLOW, cvc5_error.decode())
+            print("=" * 80)
+            print()
```

Comment:
Instead of duplicating code, would it be possible to refactor some of this into a method in the `Tester` class that can then be used by the default `run()` method as well as this `run()` method?

Reply:
Done!

Reply:
I think that there is still some opportunity to share more of the code, in particular the handling of checking the `exit_status` for `STATUS_TIMEOUT` and it being different from `EXIT_OK`. Maybe we can add a method `check_exit_status(name, expected_exit_status, exit_status)` (where `name` is either `cvc5` or `lfsc`) and then do something like:

```
exit_code = check_exit_status(name, exit_status)
if exit_code:
  return exit_code
```

Reply:
Done!